.github/workflows/reusable-python-publish_to_artifactory.yml

```diff
@@ -0,0 +1,72 @@
+name: Publish condan package to Artifactory 
+
+on:
+  workflow_call:
+    inputs:
+        package_name:
+            description: 'Name of the package to build'
+            required: true
+            type: string
+        python_ver:
+            description: 'Python version to use (eg: 3.10)'
+            required: true
+            type: string
+        version_tag:
+            description: 'Version tag of the package to build'
+            required: true
+            type: string
+    secrets:
+        EXTRA_REPO_TOKEN:
+            description: 'Bearer token for the extra repository on JFrog Artifactory'
+            required: true
+
+defaults:
+    run:
+        shell: 'bash -l {0}'
+
+jobs:
+    release_on_artifactory:
+        name: Release package on Artifactory
+        runs-on: ubuntu-latest
+        env:
+            PYTHONUTF8: 1
+            CONDA_CHANNEL_PRIORITY: strict
+            PIP_NO_DEPS: 1 # all dependencies are installed from conda
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+            -   name: Setup CONDA_LOCK_ENV_FILE
+                run: |
+                    os_conda=$( [[ $RUNNER_OS == "Linux" ]] && echo "linux" || ([[ $RUNNER_OS == "Windows" ]] && echo "win" || echo "osx"))
+                    
+                    echo "CONDA_LOCK_ENV_FILE=environments/py-${{ inputs.python_ver }}-${os_conda}-64-dev.conda.lock.yml" >> $GITHUB_ENV
+            -   uses: MiraGeoscience/CI-tools/.github/actions/reusable-python-setup_conda@main
+                name: Setup conda env
+                with:
+                  python_ver: ${{ inputs.python_ver }}
+            -   name: Install dependencies
+                run: |
+                    conda install -y grayskull
+                    conda install -y conda-build
+                    conda install -y conda-verify
+            -   name: Generate conda recipe
+                run: |
+                    tar czf ../${{inputs.package_name}}.tar.gz .
+                    grayskull pypi ../${{ inputs.package_name }}.tar.gz
```

Comment:
**question:** does not the generated recipe need to be patched up?

Reply:
We need to patch up the recipe to guarantee the description, the maintainer, etc. It can also be necessary if the package use a pip package which hasn't the same name on conda-forge. 
It's rare but not impossible. Maybe we could add the recipe inside the python package and just update the version number and the sha256, everytime we run this workflow. It could be mean no to it, instead of generate it the recipe from scratch everytime.

Reply:
indeed, it probably makes sense to commit the recipe.
I am sure this workflow will evolve as it will be put in place in the various repos